### PR TITLE
[Merged by Bors] - feat(Holder): basic lemmas for `HolderWith`

### PR DESCRIPTION
**Pull request by kex-y**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/Holder.lean [hunk 1]
```diff
@@ -190,11 +190,31 @@ theorem ediam_image_le (hf : HolderWith C r f) (s : Set X) :
...
+    HolderWith C' r f := fun x₁ x₂ => (hf x₁ x₂).trans (by
+  by_cases hC : C = 0
+  · simp only [hC, ENNReal.coe_zero, zero_mul, zero_le]
+  by_cases hx : edist x₁ x₂ ^ (r : ℝ) = ∞
+  · rw [hx, ENNReal.mul_top (coe_ne_zero.2 <| ne_of_gt <| lt_of_lt_of_le (by positivity) h)]
+    exact le_top
+  by_cases hx' : edist x₁ x₂ ^ (r : ℝ) = 0
+  · rw [hx', mul_zero]
+    exact zero_le _
+  exact (ENNReal.mul_le_mul_right hx' hx).2 <| coe_le_coe.2 h)
```
[the remark this answers is not in the document]
Comment (JasonKYi) [reply Mathlib/Topology/MetricSpace/Holder.lean]: Thanks!

File: Mathlib/Topology/MetricSpace/Holder.lean [hunk 2]
```diff
@@ -190,11 +190,23 @@ theorem ediam_image_le (hf : HolderWith C r f) (s : Set X) :
   EMetric.diam_image_le_iff.2 fun _ hx _ hy =>
     hf.edist_le_of_le <| EMetric.edist_le_diam_of_mem hx hy
 
+lemma const {y : Y} :
+    HolderWith C r (Function.const X y) := fun x₁ x₂ => by
+  simp only [Function.const_apply, edist_self, zero_le]
+
+lemma zero [Zero Y] : HolderWith C r (0 : X → Y) := .const
+
+lemma isEmpty (hX : IsEmpty X) : HolderWith C r f := fun x => False.elim <| hX.elim x
```
Comment (urkud) [inline Mathlib/Topology/MetricSpace/Holder.lean:199]: Usually we take `IsEmpty` as a typeclass assumption.
```suggestion
lemma of_isEmpty [IsEmpty X] : HolderWith C r f := isEmptyElim
```

File: Mathlib/Topology/MetricSpace/Holder.lean [hunk 3]
```diff
@@ -223,4 +235,40 @@ theorem dist_le (hf : HolderWith C r f) (x y : X) : dist (f x) (f y) ≤ C * dis
...
 end HolderWith
 
+end PseudoMetric
+
+section Metric
+
+variable [PseudoMetricSpace X] [MetricSpace Y] {C r : ℝ≥0} {f : X → Y}
+
+@[simp]
+lemma holderWith_eq_zero_iff : HolderWith 0 r f ↔ ∀ x₁ x₂, f x₁ = f x₂ := by
```
Comment (urkud) [inline Mathlib/Topology/MetricSpace/Holder.lean:245]:
```suggestion
lemma holderWith_zero_iff : HolderWith 0 r f ↔ ∀ x₁ x₂, f x₁ = f x₂ := by
```

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

**Comment by JasonKYi** [comment]
Thanks!
bors r+

**Pull request closed without merging**